source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Android/SKTouchHandler.cs

```diff
@@ -47,15 +47,32 @@ private void OnTouch(object sender, View.TouchEventArgs e)
 			var id = evt.GetPointerId(pointer);
 			var coords = scalePixels(evt.GetX(pointer), evt.GetY(pointer));
 
-			var toolType = evt.GetToolType(id);
+			var toolType = evt.GetToolType(pointer);
 			var deviceType = GetDeviceType(toolType);
 
+			var pressure = evt.GetPressure(pointer);
+
+			var button = SKMouseButton.Left;
+
+			if (toolType == MotionEventToolType.Eraser)
+			{
+				button = SKMouseButton.Middle;
+			}
+			else if (evt.ButtonState == MotionEventButtonState.StylusSecondary)
+			{
+				button = SKMouseButton.Right;
+			}
+
```

Comment:
Maybe move this into a separate method to be like the others

Reply:
will do